dashboard/grid_builder.py

```diff
@@ -115,37 +130,63 @@ def get_result(self,
         for pr in package_results:
             if not self._is_py_version_incompatible(pr) and \
                             pr.status != compatibility_store.Status.SUCCESS:
-                return {
-                    'status': pr.status.value,
-                    'self': True,
-                    'details': pr.details
-                }
+                self_result.append(
+                    {
+                        'status': pr.status.value,
+                        'self': True,
+                        'details': pr.details
+                    }
+                )
+                cell_color_status = 'Self-' + pr.status.value
 
         if package_1 == package_2:
-            return {
-                'status': compatibility_store.Status.SUCCESS.name,
-                'self': True,
-            }
+            if not self_result:
+                self_result.append(
+                    {
+                        'status': compatibility_store.Status.SUCCESS.name,
+                        'self': True,
+                    }
+                )
         else:
             pairwise_results = self._pairwise_to_results[
                 frozenset([package_1, package_2])]
             if not pairwise_results:
-                return {
-                    'status': compatibility_store.Status.UNKNOWN.name,
-                    'self': False,
-                }
+                pair_result.append(
+                    {
+                        'status': compatibility_store.Status.UNKNOWN.name,
+                        'self': False,
+                    }
+                )
+                cell_color_status = 'Pairwise-UNKNOWN'
             for pr in pairwise_results:
                 if not self._is_py_version_incompatible(pr) and \
                             pr.status != compatibility_store.Status.SUCCESS:
-                    return {
-                        'status': pr.status.value,
+                    pair_result.append(
+                        {
+                            'status': pr.status.value,
+                            'self': False,
+                            'details': pr.details
+                        }
+                    )
+                    cell_color_status = 'Pairwise-' + pr.status.value
+
+            if not pair_result:
+                pair_result.append(
+                    {
+                        'status': compatibility_store.Status.SUCCESS.name,
                         'self': False,
-                        'details': pr.details
                     }
-            return {
-                'status': compatibility_store.Status.SUCCESS.name,
-                'self': False,
-            }
+                )
+                if cell_color_status is 'Self-SUCCESS':
+                    cell_color_status = 'Pairwise-SUCCESS'
+
+        result = {
+            'color_status': cell_color_status,
```

Comment:
Are you sure that you want to split the UI logic between the Python code and the template?

Reply:
Could you explain a little more here?

Reply:
Renamed `cell_color_status` to `status_type`. Avoid mentioning the template specific stuff in `grid_builder.py`.